### PR TITLE
OPSEXP-2149 Document recipe for quay.io image tag expire

### PR DESCRIPTION
**Pull request by gionn**
### Checklist

- Jira Reference (also in PR title): OPSEXP-2149
- [README](https://github.com/Alfresco/alfresco-build-tools/blob/master/docs/README.md) updated after adding/changing behaviour of an action
- Proposed version increment for [release](https://github.com/Alfresco/alfresco-build-tools/blob/master/docs/README.md#release):
  - [ ] Patch (bugfix)
  - [x] Minor (new feature)
  - [ ] Major (breaking changes)
- External PR link where changes has been tested:

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1619,6 +1619,41 @@ variables are empty (when event is not related to a specific ref).
 
 More docs on [using concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)
 
+### Expiring tags for quay.io images
+
+It may be desirable to push docker images from branches to test them before
+merge but we should avoid to pollute image registry with tags.
```
Comment (dsibilio) [inline docs/README.md:1625]:
```suggestion
merge but we should avoid polluting the image registry with these tags.
```

File: docs/README.md [hunk 2]
```diff
@@ -1619,6 +1619,41 @@ variables are empty (when event is not related to a specific ref).
...
+
+With quay.io it can be easily achievable by setting on the docker image the label:
```
Comment (dsibilio) [inline docs/README.md:1627]:
```suggestion
With quay.io, this can be easily achieved by setting the following label on the docker image:
```

**Review by dsibilio — COMMENTED** [review]
Thanks for this! Left a few grammar checks, I would also suggest linking the official quay.io documentation that also talks about the time format to be used: https://access.redhat.com/documentation/it-it/red_hat_quay/3.2/html/use_red_hat_quay/working_with_tags#tag-expiration.

**Review by alxgomz — CHANGES_REQUESTED** [review]
Do you mind adding a quick line dedicated to fabric8 which seems to still be used in acs-packaging at least:
e.g. `MAVEN_OPTIONS` set to `$MAVEN_OPTIONS -Ddocker.labels.${{ steps.vars.outputs.image_labels }}` (https://dmp.fabric8.io/#property-configuration) or similar?

**Review by alxgomz — DISMISSED** [review]

**Review by pmacius — DISMISSED** [review]

File: docs/README.md [hunk 3]
```diff
@@ -1619,6 +1619,60 @@ variables are empty (when event is not related to a specific ref).
...
+docker image like:
+
+```properties
+quay.expires-after=2w
+```
+
+For the supported time formats, please check the [RedHat official
+documentation](https://access.redhat.com/documentation/it-it/red_hat_quay/3.2/html/use_red_hat_quay/working_with_tags#tag-expiration).
+
+An example step which compute the label could be:
```
Comment (dsibilio) [inline docs/README.md:1637]:
```suggestion
An example step which computes the label could be:
```

File: docs/README.md [hunk 4]
```diff
@@ -1619,6 +1619,60 @@ variables are empty (when event is not related to a specific ref).
...
+    if [[ "${{ github.ref_name }}" == "main" ]]; then
+      echo "image_tag=latest" >> $GITHUB_OUTPUT
+      echo "image_labels=" >> $GITHUB_OUTPUT
+    else
+      echo "image_tag=${GITHUB_REF_NAME//\//-}" >> $GITHUB_OUTPUT
+      echo "image_labels=quay.expires-after=2w" >> $GITHUB_OUTPUT
+    fi
+```
+
+Then if you are using the `docker/build-push-action` action:
```
Comment (dsibilio) [inline docs/README.md:1652]:
```suggestion
Then, if you are using the `docker/build-push-action` action:
```

**Review by dsibilio — DISMISSED** [review]

**Review by alxgomz — DISMISSED** [review]

**Review by dsibilio — APPROVED** [review]

**Review by dsibilio — APPROVED** [review]

**Review by alxgomz — APPROVED** [review]

**Pull request merged**